File: 34_Find_First_and_Last_Position_of_Element_in_Sorted_Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def result(target,left,right):
            if left>right:
                return left
            mid=(left+right)//2
            if target>=nums[mid]:
                left=mid+1
            else:
                right=mid-1
            return result(target,left,right)
        
        lower=result(target-0.5,0,len(nums)-1)
        upper=result(target+0.5,0,len(nums)-1)
        
        if lower==upper:
            return [-1,-1]
        else:
            return [lower,upper-1]
        
                
        
        
        
            
    
        

File: 34_Find_First_and_Last_Position_of_Element_in_Sorted_Array.cpp
class Solution {
public:
    int Binary_Search(vector<int> nums, int num, string occur) {
        int left = 0, right = nums.size() - 1, mid;
        int result = -1;

        while (left <= right) {
            mid = (left + right) / 2;

            if (nums[mid] == num) {
                result = mid;
                (occur == "F") ? right = mid - 1 : left = mid + 1;
            }
            else if (nums[mid] > num) {
                right = mid - 1;
            }
            else {
                left = mid + 1;
            }
        }

        return result;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
        int first = Binary_Search(nums, target, "F");
		int last = Binary_Search(nums, target, "L");
        
        return {first, lastO};
    }
    
    
};